### PR TITLE
Gun smithing update 2: Makeshift frames

### PR DESCRIPTION
**Pull request by maikilangiolo**
## About The Pull Request
Now makeshift guns properly follow the current frame assembly crafting method, and not the old, leftover generic gun parts.
While I was at it I lowered a bit the hardcase steel cost. Seriously, 20 steel? an entire locker is only 10 steel, as well as the flare shotgun which costed more steel than a rifle.
relies both on #7546 and #7530 

## Why It's Game The Good For
Obsolete bad, streamline good and smithing cooler. all the recycle parts were tweaked so they gave exactly the same materials used in crafting frame, and frame  + gun parts

## Changelog
:cl:
add: addedd new crafting recipies for all makeshift gun frames
tweak: now makeshift guns are assembled from frames and regular gun parts
balance: rebalanced ammo hardcase steel cost 20 -> 15
balance: reduced flare shotgun steel cost
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/organ_module/active/armguns.dm [hunk 1]
```diff
@@ -14,7 +14,7 @@
 	mag_well = MAG_WELL_PISTOL|MAG_WELL_H_PISTOL|MAG_WELL_SMG
 	magazine_type = /obj/item/ammo_magazine/smg
 	auto_eject = 1
-	matter = list(MATERIAL_PLASTEEL = 12, MATERIAL_PLASTIC = 3)
+	matter = list(MATERIAL_STEEL = 10, MATERIAL_PLASTEEL = 12, MATERIAL_PLASTIC = 8, MATERIAL_WOODD = 6)
```
Comment (zombiemimic) [inline code/game/objects/items/devices/organ_module/active/armguns.dm:17]:
```suggestion
	matter = list(MATERIAL_STEEL = 10, MATERIAL_PLASTEEL = 12, MATERIAL_PLASTIC = 8, MATERIAL_WOOD = 6)
```

File: code/modules/projectiles/guns/projectile/automatic/luty.dm [hunk 2]
```diff
@@ -29,12 +29,20 @@
...
-
+	gun_parts = list(/obj/item/part/gun/frame/luty = 1, /obj/item/part/gun/grip/wood = 1, /obj/item/part/gun/mechanism/smg = 1, /obj/item/part/gun/barrel/pistol = 1)
 	origin_tech = list(TECH_COMBAT = 2, TECH_MATERIAL = 2)
-	matter = list(MATERIAL_STEEL = 15, MATERIAL_WOOD = 10)
-	gun_parts = list(/obj/item/part/gun = 2 ,/obj/item/stack/material/steel = 15)
 
 
+/obj/item/part/gun/frame/luty
+	name = "Luty frame"
+	desc = "A Luty SMG. It rattles bit, but it's okay. Just dont shake it too hard."
```
Comment (zombiemimic) [inline code/modules/projectiles/guns/projectile/automatic/luty.dm:38]:
```suggestion
	desc = "A Luty SMG. It rattles bit, but it's okay. Just don't shake it too hard."
```

File: code/modules/projectiles/guns/projectile/revolver/handmade.dm [hunk 3]
```diff
@@ -15,24 +15,15 @@
...
-	..()
+/obj/item/part/gun/frame/revolver_handmade
+	name = "Handmade revolver frame"
+	desc = "A handmade revolver. The second most ancient gun design, made with scrap and spit."
+	icon_state = "frame_revolver_hm"
+	matter = list(MATERIAL_STEEL = 10, MATERIAL_PLASTIC = 5, MATERIAL_WOOD = 4)
+	result = /obj/item/gun/projectile/revolver/handmade
+	gripvars = /obj/item/part/gun/grip/wood
+	mechanismvar = /obj/item/part/gun/mechanism/revolver
+	barrelvars = list(/obj/item/part/gun/barrel/magnum, /obj/item/part/gun/barrel/pistol)
```
Comment (zombiemimic) [inline code/modules/projectiles/guns/projectile/revolver/handmade.dm:29]: newline

File: code/modules/projectiles/guns/projectile/shotgun/handmade.dm [hunk 4]
```diff
@@ -11,16 +11,27 @@
...
+	init_recoil = CARBINE_RECOIL(3.5)
 	style_damage_multiplier = 2
 	price_tag = 250 //cheap as they get
 	spawn_blacklisted = FALSE
 	spawn_tags = SPAWN_TAG_GUN_HANDMADE
+	gun_parts = list(/obj/item/part/gun/frame/ponyets = 1, /obj/item/part/gun/grip/wood = 1, /obj/item/part/gun/mechanism/shotgun = 1, /obj/item/part/gun/barrel/shotgun = 1)
+
+/obj/item/part/gun/frame/ponyets
+	name = "Ponyets frame"
+	desc = "A Ponyets. One shot, better make it count. And try not to blow your fingers off"
```
Comment (zombiemimic) [inline code/modules/projectiles/guns/projectile/shotgun/handmade.dm:28]: some sort of punctuation would be nice

**Review by zombiemimic — COMMENTED** [review]
The project seems in good hands

**Review by zombiemimic — APPROVED** [review]

**Review by hyperioo — APPROVED** [review]

**Pull request merged**